sriov: New way to wait VF been created

cathay4t opened the pull request:
The nispor 1.2.1 has include VF interface name as
`NisporSriovVf.iface_name` which will be used to check whether VF
interface been created.

The old VF name guessing method is removed. The `verify_sriov_vf()` is
created to collect all VF interfaces name, then comparing it with
`total_vfs` count and check whether they exists in current state.

cathay4t commented:
Patch still need work on decreasing SRIOV count.

cathay4t committed:
sriov: New way to wait VF been created

The nispor 1.2.1 has include VF interface name as
`NisporSriovVf.iface_name` which will be used to check whether VF
interface been created.

The old VF name guessing method is removed. The `verify_sriov_vf()` is
created to collect all VF interfaces name, then comparing it with
`total_vfs` count and check whether they exists in current state.

The test indicate kernel/udev does not require extra time to remove VF
interface when decreasing SR-IOV VF count.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
The test indicate kernel/udev does not require extra time to remove VF
interface when decreasing SR-IOV VF count.

Tested on iintel i40e card with `./sriov_test.py`.

ffmancera commented:
@cathay4t all the tests are failing on the nispor version. Could you check that? In general the changes looks good to me but I would like to see the CI test passing and try it out on a SR-IOV NIC.

ffmancera commented:
Please. @cathay4t rebase it so the test will run properly :-)

ffmancera reviewed (changes requested):
Unit tests are failing, could you check it out? Thank you! https://github.com/nmstate/nmstate/runs/4452534842?check_suite_focus=true#step:5:1590

ffmancera committed:
tests, lib: drop deprecated tests about SR-IOV VFs

We are not using the `<pf_name>v0` pattern for SR-IOV PF - VF
relationship. Therefore, these tests are now wrong.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

ffmancera reviewed (approved):
LGTM!
